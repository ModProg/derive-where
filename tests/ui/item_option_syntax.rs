use std::marker::PhantomData;

use derive_where::derive_where;

#[derive_where()]
struct EmptyAttribute<T>(PhantomData<T>);

#[derive_where = "invalid"]
struct WrongAttributeSyntax<T>(PhantomData<T>);

#[derive_where(Debug = "option")]
struct WrongOptionSyntax<T>(PhantomData<T>);

#[derive_where(Debug())]
struct EmptyOption<T>(PhantomData<T>);

#[derive_where(Debug(option))]
struct UnsupportedOption<T>(PhantomData<T>);

fn main() {}
