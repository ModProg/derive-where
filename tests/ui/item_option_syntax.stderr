error: empty `derive_where` found
 --> tests/ui/item_option_syntax.rs:5:1
  |
5 | #[derive_where()]
  | ^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the attribute macro `derive_where` (in Nightly builds, run with -Z macro-backtrace for more info)

error: key-value macro attributes are not supported
 --> tests/ui/item_option_syntax.rs:8:1
  |
8 | #[derive_where = "invalid"]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: unsupported trait syntax, expected one of Clone, Copy, Debug, Default, Deserialize, Eq, Hash, Ord, PartialEq, PartialOrd, Serialize, Zeroize, ZeroizeOnDrop
 --> tests/ui/item_option_syntax.rs:8:18
  |
8 | #[derive_where = "invalid"]
  |                  ^^^^^^^^^

error: unexpected option syntax
  --> tests/ui/item_option_syntax.rs:11:16
   |
11 | #[derive_where(Debug = "option")]
   |                ^^^^^^^^^^^^^^^^

error: empty attribute option found
  --> tests/ui/item_option_syntax.rs:14:16
   |
14 | #[derive_where(Debug())]
   |                ^^^^^^^

error: `Debug` doesn't support any options
  --> tests/ui/item_option_syntax.rs:17:16
   |
17 | #[derive_where(Debug(option))]
   |                ^^^^^^^^^^^^^
